### PR TITLE
check that item.first function exists before calling

### LINKED ISSUE #2927: Cypress.io 3.1.3 breaks by legacy jQuery library
When testing a web page with old version jQuery, Cypress.io 3.1.3 breaks, and complain a TypeError as below.

cypress-error

The trackingstack is here.
```javascript
cypress_runner.js:160760 TypeError: item.first is not a function
    at Object.unwrapFirst (http://localhost:9090/__cypress/runner/cypress_runner.js:62996:19)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:58725:29
From previous event:
    at runCommand (http://localhost:9090/__cypress/runner/cypress_runner.js:58722:8)
    at next (http://localhost:9090/__cypress/runner/cypress_runner.js:58767:14)
From previous event:
    at http://localhost:9090/__cypress/runner/cypress_runner.js:58786:37
From previous event:
    at run (http://localhost:9090/__cypress/runner/cypress_runner.js:58784:15)
    at Object.cy.(anonymous function) [as visit] (http://localhost:9090/__cypress/runner/cypress_runner.js:59012:11)
    at Context.runnable.fn (http://localhost:9090/__cypress/runner/cypress_runner.js:59142:20)
    at callFn (http://localhost:9090/__cypress/runner/cypress_runner.js:27558:21)
    at Test.Runnable.run (http://localhost:9090/__cypress/runner/cypress_runner.js:27551:7)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:62062:28
From previous event:
    at Object.onRunnableRun (http://localhost:9090/__cypress/runner/cypress_runner.js:62057:20)
    at $Cypress.action (http://localhost:9090/__cypress/runner/cypress_runner.js:57439:51)
    at Test.Runnable.run (http://localhost:9090/__cypress/runner/cypress_runner.js:61123:20)
    at Runner.runTest (http://localhost:9090/__cypress/runner/cypress_runner.js:28021:10)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:28127:12
    at next (http://localhost:9090/__cypress/runner/cypress_runner.js:27941:14)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:27951:7
    at next (http://localhost:9090/__cypress/runner/cypress_runner.js:27883:14)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:27914:7
    at next (http://localhost:9090/__cypress/runner/cypress_runner.js:62027:16)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:62039:11
From previous event:
    at onNext (http://localhost:9090/__cypress/runner/cypress_runner.js:62038:57)
    at done (http://localhost:9090/__cypress/runner/cypress_runner.js:27519:5)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:27563:11
From previous event:
    at callFn (http://localhost:9090/__cypress/runner/cypress_runner.js:27562:10)
    at Hook.Runnable.run (http://localhost:9090/__cypress/runner/cypress_runner.js:27551:7)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:62062:28
From previous event:
    at Object.onRunnableRun (http://localhost:9090/__cypress/runner/cypress_runner.js:62057:20)
    at $Cypress.action (http://localhost:9090/__cypress/runner/cypress_runner.js:57439:51)
    at Hook.Runnable.run (http://localhost:9090/__cypress/runner/cypress_runner.js:61123:20)
    at next (http://localhost:9090/__cypress/runner/cypress_runner.js:27897:10)
    at http://localhost:9090/__cypress/runner/cypress_runner.js:27919:5
    at timeslice (http://localhost:9090/__cypress/runner/cypress_runner.js:23160:27)
logError @ cypress_runner.js:160760
(anonymous) @ cypress_runner.js:160430
emitOne @ cypress_runner.js:126682
emit @ cypress_runner.js:126760
(anonymous) @ cypress_runner.js:116982
emitOne @ cypress_runner.js:126682
emit @ cypress_runner.js:126760
emit @ cypress_runner.js:117011
Test._this._onErrorClick @ cypress_runner.js:118787
(anonymous) @ cypress_runner.js:117133
executeAction @ cypress_runner.js:128987
res @ cypress_runner.js:128978
callCallback @ cypress_runner.js:133056
invokeGuardedCallbackDev @ cypress_runner.js:133106
invokeGuardedCallback @ cypress_runner.js:133159
invokeGuardedCallbackAndCatchFirstError @ cypress_runner.js:133173
executeDispatch @ cypress_runner.js:133504
executeDispatchesInOrder @ cypress_runner.js:133523
executeDispatchesAndRelease @ cypress_runner.js:133624
executeDispatchesAndReleaseTopLevel @ cypress_runner.js:133635
forEachAccumulated @ cypress_runner.js:133605
runEventsInBatch @ cypress_runner.js:133766
runExtractedEventsInBatch @ cypress_runner.js:133775
handleTopLevel @ cypress_runner.js:137768
batchedUpdates$1 @ cypress_runner.js:150409
batchedUpdates @ cypress_runner.js:135100
dispatchEvent @ cypress_runner.js:137847
interactiveUpdates$1 @ cypress_runner.js:150464
interactiveUpdates @ cypress_runner.js:135119
dispatchInteractiveEvent @ cypress_runner.js:137824
``` 

But, when I downgrade to version 3.1.2, and all tests work properly with no issues at all.

Could someone please help to have a look at this issue? Thanks

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- should help with support of jQuery legacy
- fix issue introduced by
https://github.com/cypress-io/cypress/pull/2870
- close #2927

### CODE REVIEW

**Review by brian-mann — CHANGES_REQUESTED** [review]
- needs a test, can likely use an old version of jquery as a fixture
- needs to add comments above the code referencing this issue so we understand why it's necessary

**Comment by jennifer-shehane** [comment]
So, I haven't been able to directly replicate this with the standard jQuery 2.2.4, but I've been able to replicate with the user's example code - which includes commonjs, so it's a bit of a frakenstein script, which I'd rather not do, but I added a test with this JS script if you can review and see anything problematic with it. 

But, I also think it's just common sense to check the existence of something before calling a function on it, so how harmful can the change be (famous last words).

**Review by brian-mann — COMMENTED** [review]
deleted

File: packages/driver/src/cypress/utils.coffee [hunk 1]
```diff
@@ -39,7 +39,10 @@ module.exports = {
     item = [].concat(val)[0]
 
     if $jquery.isJquery(item)
-      return item.first()
+      ## Need check to support jQuery legacy versions
+      ## https://github.com/cypress-io/cypress/issues/2927
+      if item.first
+        return item.first()
```
Comment (brian-mann) [inline packages/driver/src/cypress/utils.coffee]: We can very simply re-wrap the jquery object in our own controlled jquery object before calling `.first()`

**Review by brian-mann — CHANGES_REQUESTED** [review]
There are two primary problems:

1. this utils function must return the first DOM element in the jquery collection. simply not calling `item.first()` will cause failures in other ways since this function is no longer doing what it's supposed to be doing
2. because this PR does not replicate the actual problem, thats why all the tests are passing. we must have a test that causes the thing we're fixing to actually break

There's also one secondary problem:

1. you shouldn't commit `packages/driver/test/cypress/fixtures/jquery-2.2.4-common.js` to source control. I see that `jquery-3.2.1.js` is there, but that shouldn't be committed either. The driver can use a utility called `npm-install-version` to install multiple versions of the same library into `node_modules` which can then be used by the `.html` fixtures and avoiding the need for source control. We do this in `packages/driver/test/support/server.coffee` for react and it needs to be done for `jquery` as well - and all fixtures need to be updated to point to the specific version of jquery it needs.

File: packages/driver/src/cypress/utils.coffee (hunk 1 above)
Comment (jennifer-shehane) [reply packages/driver/src/cypress/utils.coffee]: Can we pair on this in our meeting today?

**Comment by agon21** [comment]
@jennifer-shehane I am getting one of my tests failing due to this error. `item.first is not a function`

Any planned date to release this patch on 3.1.6?
Thanks

**Review by brian-mann — APPROVED** [review]

**Pull request merged**